monai/transforms/croppad/array.py

```diff
@@ -638,11 +637,23 @@ def __call__(self, img: np.ndarray, mode: Optional[Union[NumpyPadMode, str]] = N
 class BoundingRect(Transform):
     """
     Compute coordinates of axis-aligned bounding rectangles from input image `img`.
+
+    Args:
+        select_fn: function to select expected foreground, default is to select values > 0.
     """
 
+    def __init__(self, select_fn: Callable = lambda x: x > 0) -> None:
+        self.select_fn = select_fn
+
     def __call__(self, img: np.ndarray) -> np.ndarray:
         """
-        See also: :py:class:`monai.transforms.utils.compute_bounding_rect`.
+        See also: :py:class:`monai.transforms.utils.generate_spatial_bounding_box`.
         """
-        bbox = [compute_bounding_rect(channel) for channel in img]
-        return np.stack(bbox, axis=0)
+        bbox_start = list()
+        bbox_end = list()
+
+        for channel in range(img.shape[0]):
+            start_, end_ = generate_spatial_bounding_box(img, select_fn=self.select_fn, channel_indices=channel)
+            bbox_start.append(start_)
+            bbox_end.append(end_)
+        return np.stack(bbox_start, axis=0), np.stack(bbox_end, axis=0)
```

Comment:
sorry I didn't make it clear, could this transform still return a 2D numpy array, in shape [`n_channels`, `(2*ndims)`]? currently it's 2 arrays of [`n_channels`, `ndims`]

Reply:
OK, restore back to the original return format.
Thanks.